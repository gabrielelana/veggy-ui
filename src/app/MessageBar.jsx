import React from 'react'
import PropTypes from 'prop-types'
import dispatcher from '../redux/dispatcher'
import {DismissMessage} from './globalActionNames'

const types = {
  info: 'notification is-success',
  error: 'notification is-danger',
  warn: 'notification is-warning',
}
function getTypeClass(type){
  return types[type]
}

class MessageBar extends React.Component {
  constructor() {
    super()
    this.state = { show: false }
  }
  componentWillReceiveProps(props) {
    if (props.message.length > 0){
      this.setState({show: true})
      setTimeout(() => {
        dispatcher.dispatch({type: DismissMessage, payload: {}})
        this.setState({show: false})
      }, this.props.timeout)
    }
  }
  render(){ 
    const displayStyle = { display:'none' }
    if (this.state.show){
      displayStyle.display = 'block'
    }
    const messageClass = getTypeClass(this.props.type)
    return (
      <div id="messageBox" className="column is-half is-offset-one-quarter is-gapless">
        <div className={messageClass} style={displayStyle} >
          {this.props.message}
        </div>
      </div>
    )
  }
}

MessageBar.defaultProps = {
  timeout: 4000
}

MessageBar.propTypes = {
  message: PropTypes.string.isRequired
}

export default MessageBar