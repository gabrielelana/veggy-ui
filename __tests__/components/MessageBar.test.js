import React from 'react'
import renderer from 'react-test-renderer'
import MessageBar from '../../src/app/MessageBar'

describe('<MessageBar />', () => {
  
  test('should display the message passed via props', () => {
    const component = renderer.create(<MessageBar message="We could be heroes" type="error" />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})