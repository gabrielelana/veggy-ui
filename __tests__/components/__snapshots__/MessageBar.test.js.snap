// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageBar /> should display the message passed via props 1`] = `
<div
  className="column is-half is-offset-one-quarter is-gapless"
  id="messageBox"
>
  <div
    className="notification is-danger"
    style={
      Object {
        "display": "none",
      }
    }
  >
    We could be heroes
  </div>
</div>
`;
